src/domain/core/src/utils/time_source.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::sync::Mutex;
+
+use chrono::{DateTime, Utc};
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+/// Abstracts the system time source
+pub trait SystemTimeSource: Send + Sync {
+    fn now(&self) -> DateTime<Utc>;
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+#[dill::component]
+pub struct SystemTimeSourceDefault;
+
+impl SystemTimeSource for SystemTimeSourceDefault {
+    fn now(&self) -> DateTime<Utc> {
+        Utc::now()
+    }
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct SystemTimeSourceMock {
```

Comment:
I prefer if we stick to [Gerard Meszaros](http://xunitpatterns.com/Test%20Double%20Patterns.html) classic terminilogy:
 - "mock" is something that tests behavioral impact (asserting calls, arguments, sequence)
 - "fake" is a testing-friendly alternative implementation
 - "stub"  is a test-specific object that feeds the desired indirect inputs
 
 From this POV, this is more of a "stub".

Reply:
Thanks - will rename to `SystemTimeSourceStub` in a follow up PR